### DISCUSSION TITLE
mydumper myloader and using stream

### DISCUSSION THREAD

**Discussion by sanjay92-gupta** [mydumper/mydumper#1126] [Q&A]
mydumper -v 4 -t 4 -u root --password='****' -h host1.com  -P 3306  --lock-all-tables  -o /data/in -B 'mydb'  --stream --triggers --events --routines --chunk-filesize 1024 --logfile mydumper_log.out| myloader --logfile myloader_log.out --threads=4 -u root --password='****' -h host2.com -P 3306 -v 4 -d /data/out -o -B 'mydb' --stream 

I am using an EC2 instance to take dump from one MySQL database on host1 and going to load on host2 using myloader . I see data is getting written in /data/in as specified in mydumper  , do I need to specify another dir for myloader since same /data/in is giving issues. Do I need to specify -d option for  even we are using --stream in myloader. I want to take backup and restore at the same time

**Comment by davidducos** [comment]
Hi @sanjay92-gupta, if you are using the latest prerelease, you will be able to share the same backup dir, but you need to set on mydumper side: --stream=NO_STREAM_AND_NO_DELETE. On myloader you can keep `--stream` which is going to delete the files that mydumper is not deleting.

**Reply by sanjay92-gupta** [reply]
I am using mydumper0.14.3-1. Is this one good ?

**Reply by davidducos** [reply]
Hi @sanjay92-gupta, you should review https://www.percona.com/blog/backup-and-restore-with-mydumper-on-docker/ which might help you understand better how MyDumper is working

**Comment by sanjay92-gupta** [comment]
myloader  seems like it is not doing anything since process shows sleep state 

mydumper  -t 4 -u root --password='******' -h my_host1.rds.amazonaws.com -P 3306  --lock-all-tables  -o /data/in -B 'production'  --triggers --events --routines --chunk-filesize 2048 --logfile mydumper_log.out
This ran fine and created files in /data/in and it over 1 hr

After mydumper completed, I am running
myloader --logfile myloader_log.out --threads=4 -u root --password='****' -h host2.rds.amazonaws.com -P 3306 -d /data/in -o -B 'production'
I see database and tables got created but I don't see any data and sessions shows in sleep state 

ps -ef | grep myloader
root        9597    9595  1 15:11 pts/1    00:01:54 myloader --logfile myloader_log.out

strace -p 9597
strace: Process 9597 attached
futex(0x7f53f6ffd9d0, FUTEX_WAIT, 9610, NULL^Cstrace: Process 9597 detached
 <detached ...>

Any idea what is going on here ?
